x-pack/plugins/index_lifecycle_management/server/plugin.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, Plugin, Logger, PluginInitializerContext, APICaller } from 'src/core/server';
+
+import { PLUGIN } from '../common/constants';
+import { Dependencies } from './types';
+import { registerApiRoutes } from './routes';
+import { License } from './services';
+import { IndexLifecycleManagementConfig } from './config';
+import { isEsError } from './lib/is_es_error';
+
+const indexLifecycleDataEnricher = async (indicesList: any, callAsCurrentUser: APICaller) => {
+  if (!indicesList || !indicesList.length) {
+    return;
+  }
+
+  const params = {
+    path: '/*/_ilm/explain',
+    method: 'GET',
+  };
+
+  const { indices: ilmIndicesData } = await callAsCurrentUser('transport.request', params);
+
+  return indicesList.map((index: any): any => {
+    return {
+      ...index,
+      ilm: { ...(ilmIndicesData[index.name] || {}) },
+    };
+  });
+};
+
+export class IndexLifecycleManagementServerPlugin implements Plugin<void, void, any, any> {
+  private readonly config$: Observable<IndexLifecycleManagementConfig>;
+  private readonly license: License;
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.create();
+    this.license = new License();
+  }
+
+  async setup({ http }: CoreSetup, { licensing, indexManagement }: Dependencies): Promise<void> {
+    const router = http.createRouter();
+    const config = await this.config$.pipe(first()).toPromise();
```

Comment:
I synced with @joshdover and you're right that we should be using sync lifecycle methods, but the platform team expects to introduce a sync config API. It will be easier to migrate the code to use it with the way things are currently set up, so he suggested we leave this as-is for now.